framework/src/linearfvbcs/LinearFVFunctorDirichletBC.C

```diff
@@ -0,0 +1,78 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "LinearFVFunctorDirichletBC.h"
+
+registerMooseObject("MooseApp", LinearFVFunctorDirichletBC);
+
+InputParameters
+LinearFVFunctorDirichletBC::validParams()
+{
+  InputParameters params = LinearFVBoundaryCondition::validParams();
+  params.addClassDescription(
+      "Adds a dirichlet BC which can be used for the assembly of linear "
+      "finite volume system and whose face values are determined using a functor.");
+  params.addRequiredParam<MooseFunctorName>("functor", "The functor for this boundary condition.");
+  return params;
+}
+
+LinearFVFunctorDirichletBC::LinearFVFunctorDirichletBC(const InputParameters & parameters)
+  : LinearFVBoundaryCondition(parameters), _functor(getFunctor<Real>("functor"))
+{
+}
+
+Real
+LinearFVFunctorDirichletBC::computeBoundaryValue() const
+{
+  return _functor(singleSidedFaceArg(_current_face_info), determineState());
+}
+
+Real
+LinearFVFunctorDirichletBC::computeBoundaryNormalGradient() const
+{
+  const auto elem_arg = makeElemArg(_current_face_type == FaceInfo::VarFaceNeighbors::ELEM
+                                        ? _current_face_info->elemPtr()
+                                        : _current_face_info->neighborPtr());
+  const Real distance = computeCellToFaceDistance();
+  return (_functor(singleSidedFaceArg(_current_face_info), determineState()) -
+          raw_value(_var(elem_arg, determineState()))) /
+         distance;
+}
+
+Real
+LinearFVFunctorDirichletBC::computeBoundaryValueMatrixContribution() const
+{
+  // Ths will not contribute to the matrix from the value considering that
+  // the value is independent of the solution.
+  return 0.0;
+}
+
+Real
+LinearFVFunctorDirichletBC::computeBoundaryValueRHSContribution() const
+{
+  // Fetch the boundary value from the provided functor.
+  return _functor(singleSidedFaceArg(_current_face_info), determineState());
+}
+
+Real
+LinearFVFunctorDirichletBC::computeBoundaryGradientMatrixContribution() const
+{
+  // The implicit term from the central difference approximation of the normal
+  // gradient.
+  return 1.0 / computeCellToFaceDistance();
+}
+
+Real
+LinearFVFunctorDirichletBC::computeBoundaryGradientRHSContribution() const
+{
+  // The boundary term from the central difference approximation of the
+  // normal gradient.
+  return _functor(singleSidedFaceArg(_current_face_info), determineState()) /
+         computeCellToFaceDistance();
```

Comment:
dont we have the true distance here in NLFV? instead of orthonormal

Reply:
In NL FV we use a different nonorthogonal correction. the othrogonal part (this one) can change too depending on what the correction approach.